pyramid_zipkin/zipkin.py

```diff
@@ -111,25 +128,31 @@ def zipkin_tween(handler, registry):
     :returns: pyramid tween
     """
     def tween(request):
+        # Creates zipkin attributes, attaches a zipkin_trace_id attr to the
+        # request, and pushes the attrs onto threadlocal stack.
         zipkin_attrs = create_zipkin_attr(request)
-
-        # If this request isn't sampled, don't go through the work
-        # of initializing the rest of the zipkin attributes
-        if not zipkin_attrs.is_sampled:
-            return handler(request)
-
-        # If the request IS sampled, we create thrift objects, store
-        # thread-local variables, etc, to enter zipkin logging context
-        thrift_endpoint = create_endpoint(request)
-        log_handler = ZipkinLoggerHandler(zipkin_attrs)
-        with ZipkinLoggingContext(zipkin_attrs, thrift_endpoint, log_handler,
-                                  request) as context:
-            response = handler(request)
-            context.response_status_code = response.status_code
-            context.binary_annotations_dict = get_binary_annotations(
-                    request, response)
-
-            return response
+        push_zipkin_attrs(zipkin_attrs)
+
+        # This context ensures that, regardless of what happens in the request,
+        # this set of zipkin attrs will be popped off the threadlocal stack.
+        with pop_attrs_context():
```

Comment:
I'd prefer to leave these separated. I think you were right that the contextmanager just added complexity. This is up-to-date in the latest diff.